libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/arrow/api/MapboxRouteArrowView.kt

```diff
@@ -196,4 +254,19 @@ class MapboxRouteArrowView(private val options: RouteArrowOptions) {
             style.getSourceAs<GeoJsonSource>(sourceId)?.featureCollection(featureCollection)
         }
     }
+
+    private fun rebuildLayersAndSources(style: Style, options: RouteArrowOptions) {
+        updateSource(
+            style,
+            RouteLayerConstants.ARROW_HEAD_SOURCE_ID,
+            FeatureCollection.fromFeatures(listOf())
+        )
+        updateSource(
+            style,
+            RouteLayerConstants.ARROW_SHAFT_SOURCE_ID,
+            FeatureCollection.fromFeatures(listOf())
+        )
```

Comment:
This doesn't rebuild the source, it only updates the data. We should remove and re-add a fresh instance of a source for the options to be applied, shouldn't we? Same in `MapboxRouteLineView`.

Reply:
The only option applied to the sources is tolerance.  My memory of our last conversation is that it's unlikely that option would change so there was no point removing and re-adding the sources.  Also I vaguely remember there was a technical reason but I can't remember exactly what it was.

Reply:
I don't think we can rely on likeliness. If there's a concrete option in the class, it has to be correctly re-applied. We've exposed the source tolerance as an option so we'll need to work with that.

If there's a technical blocker, let's see what that is and discuss next steps.